oci/auth/login/login.go

```diff
@@ -121,11 +120,11 @@ func (m *Manager) Login(ctx context.Context, url string, ref name.Reference, opt
 	if opts.Cache != nil {
 		key, err = m.keyFromURL(url, provider)
 		if err != nil {
-			log.Error(err, "failed to get cache key")
+			logr.FromContextOrDiscard(ctx).Error(err, "failed to get cache key")
 		} else {
-			auth, exists, err := getObjectFromCache(opts.Cache, key)
+			auth, exists, err := getObjectFromCache[authn.Authenticator](opts.Cache, key)
```

Comment:
In my IDE I get the opposite, code shows as not compiling

Reply:
I'm also on Go 1.23, this may be a Goland IDE bug

<img width="886" alt="Screenshot 2024-08-24 at 09 43 07" src="https://github.com/user-attachments/assets/7462bc82-1bd0-46c1-b9c6-7ae173746813">

Reply:
I've undone it, this is indeed an issue in IntelliJ